### PR TITLE
Fix(MIP/Scout upload): mip shared tags, revert latest file extraction for VCFs

### PR DESCRIPTION
**Pull request by ivadym**
## Description

Actual fix should be addressed here: https://github.com/Clinical-Genomics/hermes/issues/73

### Fixed
- MIP upload of index files, reverting latest file extraction for VCFs

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b [THIS-BRANCH-NAME] -a
    ```

### Tests

- [x] `cg upload create-scout-load-config <case_id>` (MIP-DNA, current issue)
Screenshot 2023-04-03 at 16 24 43
Screenshot 2023-04-04 at 11 19 25

- [x] `cg upload create-scout-load-config <case_id>` (MIP-DNA, fixed)
Screenshot 2023-04-03 at 17 21 34
Screenshot 2023-04-04 at 13 04 20

- [x] `cg upload create-scout-load-config <case_id>` (BALSAMIC, still extracting the latest filtered file 🎈 )
Screenshot 2023-04-03 at 16 42 00

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/apps/housekeeper/hk.py [hunk 1]
```diff
@@ -128,17 +128,29 @@ def files(
...
+                LOG.debug(f"Found file {file}")
                 files.append(file)
         if not files:
-            LOG.info(f"Could not find any files matching the tags {tags}")
+            LOG.warning(f"Could not find any files matching the tags {tags}")
+        return files
+
+    @staticmethod
+    def fetch_file_from_version(version: Version, tags: Set[str]) -> Optional[File]:
+        """Fetch the first file matching the given tags."""
```
Comment (henrikstranneheim) [inline cg/apps/housekeeper/hk.py:146]:
```suggestion
        """Return the first file matching the given tags."""
```

File: cg/apps/housekeeper/hk.py [hunk 2]
```diff
@@ -128,17 +128,29 @@ def files(
         )
 
     @staticmethod
-    def fetch_file_from_version(version_obj: Version, tags: Set[str]) -> Optional[File]:
-        """Return the latest file that includes at least all tags in "tags". Return None if no file could be found."""
+    def fetch_files_from_version(version: Version, tags: Set[str]) -> Optional[List[File]]:
```
Comment (henrikstranneheim) [inline cg/apps/housekeeper/hk.py:131]:
```suggestion
    def get_files_from_version(version: Version, tags: Set[str]) -> Optional[List[File]]:
```

File: tests/apps/hk/test_file.py [hunk 3]
```diff
@@ -118,7 +118,40 @@ def test_get_file(populated_housekeeper_api: MockHousekeeperAPI):
     assert hk_file is not None
 
 
-def test_fetch_file_from_version(
+def test_fetch_files_from_version(
```
Comment (henrikstranneheim) [inline tests/apps/hk/test_file.py:121]:
```suggestion
def test_get_files_from_version(
```

File: tests/mocks/hk_mock.py [hunk 4]
```diff
@@ -173,11 +173,16 @@ def __init__(self, config=None):
         self.root_path = config.get("housekeeper", {}).get("root", str(ROOT_PATH))
 
     # Mock specific functions
-    def fetch_file_from_version(self, version_obj, tags):
+    def fetch_file_from_version(self, version, tags):
```
Comment (henrikstranneheim) [inline tests/mocks/hk_mock.py:176]:
```suggestion
    def get_file_from_version(self, version, tags):
```
Add type hints

File: cg/meta/report/mip_dna.py [hunk 5]
```diff
@@ -151,3 +153,20 @@ def get_upload_case_tags(self) -> dict:
         """Retrieves MIP DNA upload case tags."""
 
         return MIP_CASE_TAGS
+
+    def get_scout_uploaded_file_from_hk(self, case_id: str, scout_tag: str) -> Optional[str]:
+        """Returns the file path of the uploaded to Scout file given its tag."""
+
+        version: Version = self.housekeeper_api.last_version(bundle=case_id)
+        tags: list = self.get_hk_scout_file_tags(scout_tag)
+        uploaded_files: Query = self.housekeeper_api.get_files(
```
[the remark this answers is not in the document]
Comment (ivadym) [reply cg/meta/report/mip_dna.py]: Yes! It returns a file iterable, I've missed that. Thanks!

**Comment by ivadym** [comment]
Screenshot 2023-04-04 at 13 36 55

Making sure the script worked, as it seems that we had problems deploying in production:
Screenshot 2023-04-04 at 13 42 25

**Pull request merged**